### PR TITLE
check pvc size before running

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
Try to fix https://github.com/pingcap/tidb-operator/issues/2734.

### What is changed and how does it work?
The pvc name for a restore job is fixed now. So when the pvc exists but the requested storage size is smaller than we set, it's better to lead users to delete the pvc.

### Check List 

Tests 

 - Unit test
 - E2E test

Code changes

 - Has Go code change

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Fail the backup job if existing PVC size is smaller than the storage request in the backup job
```

### CODE REVIEW

File: pkg/backup/backup/backup_manager.go [hunk 1]
```diff
@@ -407,9 +407,12 @@ func (bm *backupManager) ensureBackupPVCExist(backup *v1alpha1.Backup) (string,
 		return "ParseStorageSizeFailed", errMsg
 	}
 	backupPVCName := backup.GetBackupPVCName()
-	_, err = bm.pvcLister.PersistentVolumeClaims(ns).Get(backupPVCName)
+	pvc, err := bm.pvcLister.PersistentVolumeClaims(ns).Get(backupPVCName)
 
 	if err == nil {
+		if pvcRs := pvc.Spec.Resources.Requests[corev1.ResourceStorage]; pvcRs.Cmp(rs) == -1 {
+			return "", fmt.Errorf("%s/%s's backup pvc %s's storage size %s is less than expected storage size %s, please delete old pvc to continue", ns, name, pvc.GetName(), pvcRs.String(), rs.String())
```
Comment (DanielZhangQD) [inline pkg/backup/backup/backup_manager.go:414]: Add a short `reason` here.

File: pkg/backup/restore/restore_manager.go [hunk 2]
```diff
@@ -417,6 +417,8 @@ func (rm *restoreManager) ensureRestorePVCExist(restore *v1alpha1.Restore) (stri
 			errMsg := fmt.Errorf(" %s/%s create restore pvc %s failed, err: %v", ns, name, pvc.GetName(), err)
 			return "CreatePVCFailed", errMsg
 		}
+	} else if pvcRs := pvc.Spec.Resources.Requests[corev1.ResourceStorage]; pvcRs.Cmp(rs) == -1 {
+		return "", fmt.Errorf("%s/%s's restore pvc %s's storage size %s is less than expected storage size %s, please delete old pvc to continue", ns, name, pvc.GetName(), pvcRs.String(), rs.String())
```
Comment (DanielZhangQD) [inline pkg/backup/restore/restore_manager.go:421]: ditto

**Comment by DanielZhangQD** [comment]
> in addition to this, I think we should emit an event too, then users can know what happened by describing the object

The backup-manager now stores various errors in the .status.conditions, I think it's OK to keep consistent with the current behavior, WDYT? @cofyc

File: pkg/backup/backup/backup_manager.go (hunk 1 above)
Comment (lichunzhu) [reply pkg/backup/backup/backup_manager.go]: addressed in https://github.com/pingcap/tidb-operator/pull/2894/commits/f656885822777a2cc56b35542e0e13d0e539e710

**Comment by cofyc** [comment]
remove the period at the end of release note: https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md

**Comment by DanielZhangQD** [comment]
What about `Fail the backup job if existing PVC size is smaller than the storage request in the backup job` for release notes? @lichunzhu

**Comment by lichunzhu** [comment]
> Fail the backup job if existing PVC size is smaller than the storage request in the backup job

SGTM, revised.

**Pull request merged**